Fix Web asking for clipboard permissions

justinmc opened the pull request:
## Description

https://github.com/flutter/flutter/pull/57139 checked the clipboard to see if it could show the paste button, but this broke web because it has to ask for the user's permission to use the clipboard (see https://github.com/flutter/flutter/pull/57139#issuecomment-629048058).  Web doesn't draw its own text selection menu though, so it doesn't need to check the clipboard, and this PR removes that functionality for web.

## Related Issues

https://github.com/flutter/flutter/pull/57286

## Tests

I added the following tests:

  * Check that web doesn't ask for clipboard contents when selecting text, but other environments do.

justinmc committed:
Dont check clipboard status on web. Unnecessary because no Flututer selection menu.

justinmc commented:
@nturgut Is it possible to run those failing web tests locally to make sure that this fixes it?

nturgut commented:
We also have this control for the toolbar: https://github.com/flutter/flutter/blob/8de07d5527bcdc6b02e43e8efed19219a84bf82e/packages/flutter/lib/src/widgets/editable_text.dart#L1932

Is it related?

For a text field on the web, we don't render a copy/paste menu, we use the one that browser shows.

justinmc commented:
Yeah, these changes in EditableText are still needed in addition to that check because EditableText uses the clipboard status to decide to allow pasting via semantics/accessibility.

goderbauer reviewed (approved):
LGTM

justinmc committed:
Revert "Revert " Bring back paste button hide behavior 3 (#57139)" (#57286)"

This reverts commit 32547dcc7ee36a9d387dbf2d111b21a12b8192be.

justinmc commented:
I just rebased with master and reverted https://github.com/flutter/flutter/pull/57286 as part of this PR as well.  I tried running a web app before and after https://github.com/flutter/flutter/pull/57324/commits/6a4c68a0247374de7f2b4417573416afb4836567 and confirmed that I saw the permissions notification only before that commit.

nturgut commented:
One option (for local development) is using [felt](https://github.com/flutter/engine/blob/master/lib/web_ui/dev/README.md) tool.

You can run:
```
felt test --integration-tests-only --use-system-flutter
```

nturgut commented:
Flutter Web integration tests are passing, I run them on my local against this PR.

justinmc commented:
Confirmed that those tests pass locally now ✅
